docs/glossary.md

```diff
@@ -0,0 +1,258 @@
+# AR.IO Node Glossary
+
+This glossary defines key terms and concepts used throughout the AR.IO Node
+codebase. Terms are organized by category and alphabetically within each
+section.
+
+## Core Arweave Concepts
+
+**Block** - The fundamental unit of the Arweave blockchain containing a set of
+transactions. Each block has a unique height (sequential number) and hash
+identifier.
+
+**Chunk** - A 256KB segment of transaction data. Large transactions are split
+into chunks for efficient storage and retrieval using Merkle trees for
+verification.
+
+**Data Root** - The Merkle tree root hash of a transaction's data chunks. Used
+to cryptographically verify data integrity without downloading the entire
+transaction.
+
+**Stable/New Data** - Classification based on block confirmations. Stable data
+is from blocks unlikely to be reorganized (18+ blocks deep), while new data is
+from recent blocks that could still be affected by chain reorganizations.
+
+**Tags** - Key-value metadata pairs attached to transactions and data items.
+Tags are indexed and searchable, enabling content discovery and
+application-specific functionality.
+
+**Transaction** - A data storage unit on Arweave identified by a unique
+43-character base64url ID. Contains data, tags, and cryptographic signatures
+proving ownership and integrity.
+
+## ANS-104 Bundle System
+
+**ANS-102/ANS-104** - Arweave standards for bundling multiple data items into a
```

Comment:
One downside of going in alphabetical order is that foundational knowledge can't be built up linearly as you read. It's not hard to beat that shortcoming though with some light linking. If we could mark this up in such a way that a term like "data items" could be referred to in a linked manner, that could help. At worst, something like "see 'data item' for more info" could potentially help.